src/services/completions.ts

```diff
@@ -2210,6 +2210,8 @@ namespace ts.Completions {
 
     function isValidTrigger(sourceFile: SourceFile, triggerCharacter: string, contextToken: Node, position: number): boolean {
         switch (triggerCharacter) {
+            case ".":

```

Comment:
What if `triggerCharacter` was a closed union of literals to ensure we handle them appropriately?